firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -107,6 +114,11 @@ public InstallationResponse createFirebaseInstallation(
         httpsURLConnection.setRequestMethod("POST");
         httpsURLConnection.setDoOutput(true);
 
+        // Set the iid token header for authenticating the iid migration to FIS.
+        if (iidToken != null) {
+          httpsURLConnection.addRequestProperty(X_ANDROID_IID_MIGRATION_KEY, iidToken);
+        }
```

Comment:
@andirayo The check when setting the iid-token header.

Reply:
I think, the check should be the other way around:
If we do not have an IID-Token and it's an Instance-ID then we don't even need to send the request, haha.
I'm fine with it, I just think the logic is a bit backwards.
You are always sending an IID-Token if available even if the identifier is an FID. Vice versa you might send migration requests for Instance-IDs without IID-Token.
I don't know if this is worth fixing, but I wanted to mention it.
If you don't fix it, please document
a) that this is the case
b) and why you decided to implement it this way